internal/compiler/object_tree.rs

```diff
@@ -469,6 +466,9 @@ pub struct Element {
     pub bindings: BindingsMap,
     pub property_analysis: RefCell<HashMap<String, PropertyAnalysis>>,
 
+    /// Code to be inserted into the constructor for root elements of components
```

Comment:
Why do we actually restrict of root elements?
Also, within repeater, it is not the root element.  (even if a "component" will be created later, this is not conceptually a component)

Reply:
That's true (the repeaters part). I mean, yeah, we could also allow `init => {}` anywhere. I'll adjust the comment in this PR to describe the reality, and we can discuss the new scope in the other PR.